### ISSUE TITLE
mindLAMP Audio Data Not Retrievable

### ISSUE THREAD

**Issue by ertjlane** [BIDMCDigitalPsychiatry/LAMP-platform#766]
**Describe the bug**
When pulling .json files for ema and audio diary data there is a change where the diaries are listed. There is no longer a url field under static_data field, which is where the sound_0, sound_1, etc. info used to be stored for repeated submissions in a single day. This problem appears to have started between 4/13 and 4/15. Notably, it was tested if audio data from before those dates could be re-extracted and it could not. 

Code used to extract data: 

import LAMP

access_key = 'admin'
secret_key = 'XXX'
api_url = 'mindlamp-api.pronet.med.yale.edu'
LAMP.connect(access_key, secret_key, api_url)

# _from & to below is for 04/09
subject_id = 'U6435702683'
activity_events_dicts = LAMP.ActivityEvent.all_by_participant(
        subject_id, _from=1680998400000, to=1681084800000,
        _limit=1000000)
print(activity_events_dicts)

# activity_events_dicts does not contain ['static_data']['url'] which used to contain the mp3 data

**Comment by ertjlane** [comment]
@sarithapillai8 Any updates on this?

**Comment by ZCOEngineer** [comment]
@ertjlane saritha is out sick, we shall update this asap.

**Comment by sarithapillai8** [comment]
@ertjlane 
![image](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/assets/63710347/007f1e17-c27f-4d55-826e-c1516c581dd7)

![image](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/assets/63710347/5ab4ca23-ad85-4e85-8409-5b58b48f024d)

The audio data is being updated to activity events. But they are not returning because we are deleting this data from the object before returning it due to performance issues. Please see the code: https://github.com/BIDMCDigitalPsychiatry/LAMP-server/blob/master/src/repository/couch/ActivityEventRepository.ts#L38

As server change is required, please confirm.

@avaidyam

**Comment by avaidyam** [comment]
Isn't there a flag for including binary data? Why are we not respecting that flag?

**Comment by sarithapillai8** [comment]
@avaidyam I was expecting that flag, but ignore_binary is used for activities, not for events.
https://github.com/BIDMCDigitalPsychiatry/LAMP-server/blob/master/src/repository/couch/ActivityRepository.ts#L19

Can we change it to support this flag?

**Comment by avaidyam** [comment]
Yes please add that flag to Activity, Sensor, ActivitySpec, SensorSpec, ActivityEvent, and SensorEvent.

**Comment by ertjlane** [comment]
Thanks for working on this - do we know why/how this switch happened, as I understand audio data would return when retrieved several months ago.

**Comment by sarithapillai8** [comment]
@ertjlane
Please see this commit: https://github.com/BIDMCDigitalPsychiatry/LAMP-server/commit/be23725a89fe65dc938e655a56666e071966f67e done by @falmeida-orangeloops

**Comment by ertjlane** [comment]
@sarithapillai8 What is the status of this?

**Comment by michaelmenon** [comment]
thanks @ertjlane , Team will reply with status update asap.

**Comment by sarithapillai8** [comment]
@ertjlane we are working on #764 as it is also a priority task and we planned to look into this after that. Only balloon risk changes are pending and we are still working onto get  the values to match with standard deviation and mean. We can hold that one and start working on this. Sorry for the late response.

**Comment by sarithapillai8** [comment]
https://api-staging.lamp.digital/participant/<participantId>/activity_event?ignore_binary=true

'ignore_binary' flag added for activity events in api-staging. Can you suggest which field needs to be controlled with this flag for sensor events?

@ertjlane

**Comment by ertjlane** [comment]
Hello @sarithapillai8, I believe you should defer to the specifications provided by Aditya above:  "please add that flag to Activity, Sensor, ActivitySpec, SensorSpec, ActivityEvent, and SensorEvent."

**Comment by sarithapillai8** [comment]
@ertjlane I have added the flag to all those specified by Aditya. But I am not sure which field should be considered for this condition for sensor-related data. Please review the changes https://github.com/BIDMCDigitalPsychiatry/LAMP-server/pull/232

**Comment by ertjlane** [comment]
@avaidyam  Do you have additional insight into this?

**Comment by ertjlane** [comment]
We can go ahead and merge any changes that have already been made, to resolve this promptly. We will circle back around to the issue of sensor fields at a later time for now lets push the changes and release.

**Comment by sarithapillai8** [comment]
We have merged the PR. The release is normally done by your team. So could you confirm that we have to do the release?
@ertjlane

**Comment by ertjlane** [comment]
As a point of clarification, given that audio data wasn't retrievable, where would the backup be? On the mindLAMP servers collecting the data, or on users phones?

**Comment by sarithapillai8** [comment]
@ertjlane Audio data is there in the database. The API was filtering the audio data in response. Now API code has changed so that audio data is filtering is now working based on ignore_binary condition.

**Comment by ertjlane** [comment]
The issue continues to be resolved, but I have a follow up query. Prior to the emergence of the issue, the audio diary URL  field contained short strings like SOUND_0 , SOUND_1 , so it used to look something llike [{‘static_data’: {‘url’: ‘SOUND_0’, `SOUND_1` …}}]. Would it be possible to return the files collected during this time frame to this naming convention or no?

**Comment by ertjlane** [comment]
@sarithapillai8 Following up on this ^

**Comment by sarithapillai8** [comment]
@ertjlane I am not sure about this. We were not following this kind of naming convention. We are saving binary audio data in this field. If these are old data, then they might be inserted as test data at the initial stage.
